components/lvgl_esp32_drivers/lvgl_tft/ili9488.c

```diff
@@ -121,7 +117,11 @@ void ili9488_flush(lv_disp_drv_t * drv, const lv_area_t * area, lv_color_t * col
     uint32_t size = lv_area_get_width(area) * lv_area_get_height(area);
 
     lv_color16_t *buffer_16bit = (lv_color16_t *) color_map;
-    uint8_t *mybuf = (uint8_t *) heap_caps_malloc(3 * size * sizeof(uint8_t), MALLOC_CAP_DMA);
+    uint8_t *mybuf;
+    do {
+        mybuf = (uint8_t *) heap_caps_malloc(3 * size * sizeof(uint8_t), MALLOC_CAP_DMA);
+        if (mybuf == NULL)  ESP_LOGW(TAG, "Could not allocate enough DMA memory!");
+    } while (mybuf == NULL);
```

Comment:
How would more memory become available in a realistic scenario? This looks like it could end up spinning in a loop indefinitely. Perhaps an `abort()` or the ESP equivalent should be used instead?

Reply:
The SPI code was reworked because @davidjade detected some potential memory leaks #112, may be the `do {} while` loop is not longer needed, I will test it later with a ILI9488 display.

Reply:
I agree that I don't see how this would really work once it fails to allocate unless there is some way to free DMA capable memory to retry. But it is the right thing to do, to allocate DMA capable memory in anything that gets handed off to a SPI request. 

This gets really trick though as that allocated DMA memory is now tied up in a SPI transaction and cannot be freed until the transaction results are retrieved/completed. I did not look into this particular driver so I don't know if it does that. If it doesn't, then it likely has to be freed in the SPI interrupt handler or some other specific driver callback upon SPI transaction completion. Or maybe held until the next time and reused/resized.

It would probably be best to not do any DMA allocations if possible and use a static buffer like the other drivers (or maybe just allocate once and reuse it).

Reply:
Thanks, I will look into using a statically allocated memory chunk.